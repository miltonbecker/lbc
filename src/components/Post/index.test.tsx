import React from 'react';
import renderer from 'react-test-renderer';
import { Post } from './';

describe('Post', () => {
  it('Renders correctly when logged in', () => {
    const tree = renderer
      .create(
        <Post
          textRef={React.createRef()}
          handlePost={e => {}}
          confidential={true}
          handleConfidentialChange={() => {}}
          loggedIn={true}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly when not logged in', () => {
    const tree = renderer
      .create(
        <Post
          textRef={React.createRef()}
          handlePost={e => {}}
          confidential={false}
          handleConfidentialChange={() => {}}
          loggedIn={false}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
