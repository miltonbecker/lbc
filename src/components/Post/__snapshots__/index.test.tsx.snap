// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Renders correctly when logged in 1`] = `
<div
  className="post"
>
  <h2>
    Add message
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <textarea
      cols="50"
      rows="6"
    />
    <label
      className="checkbox"
    >
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
       
      Confidential
    </label>
    <button>
      Post
    </button>
  </form>
</div>
`;

exports[`Post Renders correctly when not logged in 1`] = `
<div
  className="post"
>
  <h2>
    Add message
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <textarea
      cols="50"
      rows="6"
    />
    <button>
      Post
    </button>
  </form>
</div>
`;
